docs/database/sql-server.md

```diff
@@ -152,6 +154,29 @@ and SHOULD be provided **at span creation time** (if provided at all):
 <!-- END AUTOGENERATED TEXT -->
 <!-- endsemconv -->
 
+### Propagating context to SQL Server
+
+**Status**: [Development][DocumentStatus]
+
+Instrumentations SHOULD propagate the context information to the SQL queries following [sqlcommenter](https://google.github.io/sqlcommenter/spec/).
+
+#### Recommended sqlcommenter attributes
+
+| Attribute              | Type   | Description                           | Require level               | Stability                                                      |
+|------------------------|--------|---------------------------------------|-----------------------------|----------------------------------------------------------------|
+| `baggage.service.name` | string | Logical name of the service [1]       | `Conditionally Required`[2] | ![Development](https://img.shields.io/badge/-development-blue) |
+| `traceparent`          | string | The trace context of current span [3] | `Recommended`               | ![Development](https://img.shields.io/badge/-development-blue) |
+
+**[1] `baggage.service.name`:** MUST be the same for all instances of horizontally scaled services. If the value was not specified, SDKs MUST fall back to `unknown_service:` concatenated with [process.executable.name](https://opentelemetry.io/docs/specs/semconv/attributes-registry/process/), e.g. `unknown_service:bash`. If `process.executable.name` is not available, the value MUST be set to `unknown_service`.
+
+[Baggage](https://www.w3.org/TR/baggage/) means you can pass data across services and processes, making it available to add to traces, metrics, or logs in those services.
+
+**[2] `baggage.service.name`:** SHOULD be set if `traceparent` cannot be injected due to performance issues.
+
+**[3] `traceparent`:** MUST be in the [text format](https://www.w3.org/TR/trace-context/#traceparent-header).
+
+Instrumentations SHOULD make use of [SET CONTEXT_INFO](https://learn.microsoft.com/en-us/sql/t-sql/statements/set-context-info-transact-sql?view=sql-server-ver16) to add high-cardinality information(e.g. `traceparent`) as adding sql comments to queries changes their unique identifier in SQL Server.
```

Comment:
We should add to the guidance here that instrumentations will need to copy over the transaction (if present) to the new `set context_info` command. When we didn't do this, we ran into customer issues when there was an active transaction

Reply:
Could you elaborate more about what `copy over the transaction` is actually doing? Could we just run `set context_info` before every query?

Reply:
Here's the LOC I'm referencing for our C# client : https://github.com/DataDog/dd-trace-dotnet/pull/6233/files#diff-29ead36b0177265c725705ce75f5a441417f4c22e99f7080bc8ef0f486b723a7R145-R146

When creating a new `set context_info` command that we run before the original query, we simply copy the `Transaction` from the previous `DbCommand` object to the new one. Otherwise, we encountered a `InvalidOperationException`

Reply:
`SET context_info` must run on the same physical connection as the SQL statement it’s meant to instrument.

Reply:
Updated